CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+## 5.0.0 - 2020-09-02
+
+### Changed
+
+We have replaced the use of the npm [request-promise](https://www.npmjs.com/package/request-promise) package with [axios](https://www.npmjs.com/package/axios) as the npm `request` package has been deprecated. This makes the following breaking changes.
+
+1. The `object` returned by an API call is now of the form of an [axios response](https://www.npmjs.com/package/axios#response-schema) which has a different interface to a [request response](https://nodejs.org/api/http.html#http_class_http_serverresponse), of particular note
```

Comment:
I think a `request` response is an instance of [http.IncomingMessage](https://nodejs.org/api/http.html#http_class_http_incomingmessage), not [http.ServerResponse](https://nodejs.org/api/http.html#http_class_http_serverresponse).

I think this by looking at `response.constructor.toString()`.

I also think you can tell this from reading the `request` docs but not easily. Here's why I think that:

The normal use of `request` involves sending a callback function to a GET request. I'm assuming request-promise then passes the same arguments the callback gets to its [onFulfilled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Parameters) callback when it resolves the promise (which is then passed to any `.then`'s chained off that promise).

The [request docs](https://github.com/request/request#requestoptions-callback) list the arguments sent to the callback as:

"The callback argument gets 3 arguments:
- An error when applicable (usually from http.ClientRequest object)
- An http.IncomingMessage object (Response object)
- The third is the response body (String or Buffer, or JSON object if the json option is supplied)"

Reply:
Good spot Tom!

So I think you are correct that `request` returns an `http.IncomingMessage` object. I've also verified this by doing 

```
notifyClient
  .getNotificationById("cdfc61f4-9408-4b8a-87ac-574da1d6a575")
  .then((response) => console.log(response.constructor.name))
  .catch((err) => console.error(err))

# output
IncomingMessage
```

One thing I am confused about is that I can see that for a response object, of class `http.IncomingMessage` that it has a property `statusCode` as is referenced in some of integration tests. https://nodejs.org/api/http.html#http_message_statuscode

However, when I look at the docs for `http.IncomingMessage`, I can't see a property `body` on but in our integration tests we regularly access `response.body`. Where is this property coming from or why can't I see it in the docs?

I can't see any transformation of the response object being done in `request-promise` either which I wondered if it was adding `body` as a property.

Any ideas?

Reply:
See latest commit added for the fix for this (although I still don't know where `response.body` is coming from)

Reply:
I found https://stackoverflow.com/questions/36948231/nodejs-request-module-body-on-http-incomingmessage which mentions `The response object passed to the response event is a standard http.IncomingMessage and as such has no body property.` although I can't find some code to underpin that

Reply:
I think `response.body` is a convenience property, used internally by request to hold the output of the `response` (http.IncomingMessage) stream. I think this because, in the `request` code:
- a method called readResponseBody pipes the `response` into a string and stores that in`response.body` (see https://github.com/request/request/blob/master/request.js#L1098)
- readResponseBody emits a 'complete' event when done which calls the callback with these arguments: `response`, `response.body` (see https://github.com/request/request/blob/master/request.js#L188)

Those arguments are listed in the docs (and my first comment of this thread as `response` and `body`

Reply:
Thanks Tom. Good find.

So it looks like in conclusion, it is an instance of `http.IncomingMessage` but they have also added a convenience property `body` on to it that is therefore not in the docs for `http.IncomingMessage`.

Are you happy for us to keep the reference
`to a [request response](https://nodejs.org/api/http.html#http_class_http_incomingmessage)` in the changelog? It is technically correct that it is an `http.IncomingMessage`, I'm just slightly cautious of users getting confused when the convenience property isn't documented by that link. I think it's acceptable, just not perfect.

Reply:
Yep, let's keep it as straightforward as possible 👍